lib/slave/slave.go

```diff
@@ -11,27 +11,31 @@ import (
 
 var slaveInstance Slave
 
+type task struct {
+	lib.Task
+	Status       lib.Status
+	Password     string
+	progressChan chan chan string
+}
+
 type Slave struct {
+	id              string
```

Comment:
Well, I think we're not gonna do that anymore, the slaves are just identified by their IP address. There's no need for slave themselves to know the IP address.